File: 011.funcaocomreturn/funcao22.erl
-module(funcao22).

-compile(export_all).

-import(loop, [for/3, while/2]).

quadrado(Var_a1) ->
    begin
      Var_temp1 = Var_a1 * Var_a1,
      st:insert({"temp", Var_temp1})
    end,
    Var_temp1.

cubo(Var_a1) ->
    begin
      Var_a2 = Var_a1 * (Var_a1 * Var_a1),
      st:insert({"a", Var_a2})
    end,
    io:format("O valor de a ao cubo: "),
    io:format("~p~n", [Var_a2]).

main(Var_Args1) ->
    st:new(),
    begin Var_x1 = 5, st:insert({"x", Var_x1}) end,
    begin
      Var_t1 = quadrado(Var_x1), st:insert({"t", Var_t1})
    end,
    cubo(Var_x1),
    io:format("O valor de a ao quadrado eh: "),
    io:format("~p~n", [Var_t1]).

File: java_src/demostracao/Funcao22.java
//package funcao22;

public class Funcao22 {

		public static boolean bo(boolean a) {
			boolean b;
			b = false;

			if (a == true)
				b = false;
			else b = true; 

			return b;
		}		
			
		public static int quadrado(int a) {
			int temp;

			temp = a*a;

			return temp;
		}

		public static void cubo(int a) {
			a = a*a*a;
			System.out.print("O valor de a ao cubo: ");
			System.out.println(a);
		}

		public static void main(String[] Args) {
			int x;
			int t;
			boolean b;

			x = 5;
			t = quadrado(x);
			cubo(x);
			System.out.print("O valor de a ao quadrado eh: ");
			System.out.println(t);
			
			b = bo(true);

			System.out.print("O valor de b ao quadrado eh: ");
			System.out.println(b);
		}
}
